core/storage/user/gae_models.py

```diff
@@ -97,6 +97,85 @@ def get_by_normalized_username(cls, normalized_username):
             cls.normalized_username == normalized_username).get()
 
 
+class ActivitiesCompletedByLearnerModel(base_models.BaseModel):
+    """Keeps track of all the explorations and collections completed by the
+    learner.
+
+    Instances of this class are keyed by the user id.
+    """
+    # IDs of all the explorations completed by the user.
+    completed_exploration_ids = ndb.StringProperty(repeated=True, indexed=True)
```

Comment:
We're running into some issues in other files with lines being too long. I wonder whether it would make sense to call the properties here just exploration_ids and collection_ids (and ditto in the domain layer), given that the model is already an "ActivitiesCompleted" model?